---
title: "site quality exploration"
output:
  html_document:
    code_folding: hide
    theme: cerulean
    toc: yes
    toc_depth: 5
    toc_float:
      collapsed: yes
      df_print: paged
editor_options: 
  chunk_output_type: console
  markdown: 
    wrap: 72
---

```{=html}
<style type="text/css">
  body{
  font-size: 12pt;
}
</style>
```

```{r libraries, echo=FALSE, eval=TRUE}
library(vcfR)
library(tidyverse)
library(ggokabeito) 
library(RcppRoll)
library(stringr)
library(purrr)
library(patchwork)
```

```{r load data}
# Load the metadata CSV file into a data frame 'dat', and keep only one F1 female per family. 
dat <- read_csv("/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/meta_data_223.csv") %>% 
   rename(id = sample) %>% # Rename 'sample' column to 'id'
 filter(!str_detect(name, "sis")) %>% # remove sisters and aunties, F1s without offspring
  filter(!str_detect(name, "479-2")) %>%   # in family 478, i kept 479-1
    filter(!str_detect(name, "565-2")) %>%   # in Family 564, I kept 565-1
    filter(!str_detect(name, "535_1")) %>%   # in Family 534, I kept 535-2
    filter(!str_detect(name, "535_3"))  # in Family 534, I kept 535-2


# Read the VCF file into an object 'vcf' using the vcfR library; 'verbose = FALSE' suppresses extra output
vcf <- read.vcfR("/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/vcf_filter/Q40BIALLDP16HDP40mis.5Chr7/Q40BIALLDP16HDP40mis.5Chr7.recode.vcf", verbose = FALSE)

# Display a summary of the VCF object (basic details about the VCF data)
vcf

# Rename the column names of the genotype matrix ('vcf@gt'):
colnames(vcf@gt) <- sub("_S\\d+$", "", colnames(vcf@gt))
  
# Extract the genotype (GT) and depth-related fields from the FORMAT column
gt <- extract.gt(vcf, element = "GT", as.numeric = FALSE)  # Extract Genotype (GT)
dp <- extract.gt(vcf, element = "DP", as.numeric = TRUE)  # Extract Depth (DP)
ao <- extract.gt(vcf, element = "AO", as.numeric = TRUE)  # Extract Alternate Observations (AO)
ad <- extract.gt(vcf, element = "AD") # Extract the allele depth (AD)
```

## Variant Allele Frequency (VAF) 
```{r}
# Ensure AO and DP are numeric and avoid division by zero
ao[is.na(ao)] <- 0
dp[is.na(dp)] <- 1  # Avoid division by zero

# Calculate VAF for each site and sample
vaf_df <- ao / dp

# Convert to a long format for easier visualization
vaf_long <- as.data.frame(vaf_df) %>%
  rownames_to_column(var = "site") %>%
  pivot_longer(-site, names_to = "id", values_to = "VAF")
```

## Genotyping quality (QAF)
```{r}
qa <- extract.gt(vcf, element = "QA", as.numeric = TRUE)  # Quality of Alt Allele
qo <- extract.gt(vcf, element = "QR", as.numeric = TRUE)  # Quality of Ref Allele

# Ensure missing values are handled
qa[is.na(qa)] <- 0
qo[is.na(qo)] <- 0

# Compute QAF for each site and sample
qaf <- qa / (qa + qo)

# Convert to long format for better visualization
qaf_long <- as.data.frame(qaf) %>%
  rownames_to_column(var = "site") %>%
  pivot_longer(-site, names_to = "id", values_to = "QAF")
```

## Heterozygosity
```{r}
is_het <- as.data.frame(is_het(gt)) %>% 
  rownames_to_column(var = "site") %>% 
  pivot_longer(-site, names_to = "id", values_to = "het") 
```

## Allele depth (AD)
```{r}
ad2 <- as.data.frame(ad) %>% 
  rownames_to_column(var = "site") %>% 
  pivot_longer(-site, names_to = "id", values_to = "ad") %>% 
  separate("ad", into = c("ref","alt"), sep = ",") 

ad_join <- select(dat, generation, sex, family, id = name) %>% 
  left_join(ad2) %>%  
  mutate(chrom = str_split(site, pattern = "_", simplify = TRUE)[, 2]) 

ad_join <- left_join(is_het, ad_join)
ad_join <- ad_join %>% mutate(ref = as.numeric(ref), alt = as.numeric(alt)) 
ad_join <- mutate(ad_join, chrom = factor(chrom, levels = unique(chrom), labels = seq_len(7)))  
```

## Genotype (GT)
```{r}
gt_long <- gt %>% 
as.data.frame() %>% 
  rownames_to_column("site") %>% 
  pivot_longer(cols = -site, names_to = "id", values_to = "GT")
```

## Depth (DP)
```{r}
dp_long <- dp %>% 
as.data.frame() %>% 
  rownames_to_column("site") %>% 
  pivot_longer(cols = -site, names_to = "id", values_to = "DP")
```

## Join all site parameters
```{r}
# Combine information on allele depth, heterozygosity, VAF, GQ and family structure 
all_param <- ad_join %>%
  left_join(qaf_long, by = c("site", "id")) %>%
  left_join(vaf_long, by = c("site", "id")) %>%
  left_join(gt_long, by = c("site", "id")) %>%
  left_join(dp_long, by = c("site", "id")) %>%
na.omit() %>%
  rename(Sample = id) %>% 
mutate(member = case_when(
    generation == "F0" & sex == "female" ~ "grandmother",
    generation == "F1" & sex == "male" ~ "father",
    generation == "F1" & sex == "female" ~ "mother",
    generation == "F2" & sex == "male" ~ "son",
    generation == "F2" & sex == "female" ~ "daughter",
    TRUE ~ "unknown"
  )) %>%
  separate(site, sep = ".1_", into = c("Chrom", "pos"), remove = FALSE) %>%  # Keep "site"
mutate(pos = as.numeric(pos))

all_param <- all_param %>% filter(!(member %in% "unknown")) %>% # remove nymphs, keep only adult mites with determined sex 
 mutate(
    GT = factor(GT, levels = c("0/0", "0/1", "1/1"), labels = c("AA", "AB", "BB")),  # Ensure "BB" exists in factor levels
    member = factor(member, levels = c("grandmother", "father", "mother", "son", "daughter"), ordered = TRUE)  # Explicit order
  ) %>%
  mutate(member_num = as.numeric(member))  # Convert ordered factor to numeric
```

# Sanity check, to determine the threshold for site filtering
In this section, we will filter the sites based on the VAF and QAF distributions, under the assumption, that when F0 and F1 are all homozygous, then their F2 female offspring, must be homozygous as well.    
As a result, we can also assume, that any genotype in the F2 female that is NOT AA, is an error.    
It is therefore should be safe to set the threshold for site quality (VAF and QAF values), based on these filtered sites.   

## F0 and F1 Homo AA, looking at F2 daughter (should all be AA)
### Find a threshold for VAF and QAF 
```{r}
# Find positions where all F0 and all F1 individuals are homozygous AA
AA_sites_F0F1 <- all_param %>%
  filter(generation %in% c("F0", "F1")) %>%  # Keep only F0 and F1 individuals
  group_by(family, site) %>%  # Group by family and position
  summarise(all_homo = all(GT == "AA"), .groups = "drop") %>%  # Ensure all individuals in family are AA
  filter(all_homo) %>%  # Keep only positions where all individuals in F0 and F1 are AA
  select(family, site)  # Extract family and positions

# Count the number of positions per family that passed the homo filtering
AA_sites_F0F1 %>%
  group_by(family) %>%
  summarise(num_sites_passed = n())
 
# Filter the F2 females at the selected positions, on chrom 1
f2_homo_parents <- all_param %>%
  filter(generation == "F2") %>%
  filter(chrom == "1") %>%
  filter(sex == "female") %>%
  # Join with AA_sites_F0F1 to keep only sites that passed the filtering
  inner_join(AA_sites_F0F1, by = c("family", "site"))

# Visualize VAF and QAF distributions to determine thresholds
# QAF: All genotypes overlaid
QAF_all_gt <- ggplot(f2_homo_parents, aes(x = QAF, fill = GT)) +
  geom_histogram(binwidth = 0.02, alpha = 1, position = "identity") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.1)) +
  theme_classic() +
  labs(title = "A",
       x = "Quality Allele Frequency (QAF)",
       y = "Count") +
  scale_fill_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),
    name = "Genotype"
  ) +
  theme(legend.position = "none")

# QAF: Facetted by genotype
QAF_facet_gt <- ggplot(f2_homo_parents, aes(x = QAF, fill = GT)) +
  geom_histogram(binwidth = 0.02, alpha = 1, position = "identity") +
  facet_wrap(~ GT, scales = "free_y") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.1)) +
  theme_classic() +
  labs(title = "B",
       x = "Quality Allele Frequency (QAF)",
       y = "Count") +
  scale_fill_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),
    name = "Genotype"
  ) +
  theme(legend.position = "right")

# Combine QAF plots
QAF_combined <- (QAF_all_gt + QAF_facet_gt + plot_layout(widths = c(1, 3)))

# Display the combined QAF plot
QAF_combined

# Add caption below the plot
cat("**Figure 1.** Distribution of Quality Allele Frequency (QAF) values in F2 females from crosses with homozygous F0 and F1 parents.\nA. Overlaid histogram of QAF values for all genotypes.\nB. Same data faceted by genotype (AA, AB, BB), showing the distribution and spread within each genotype class.\nVertical axis represents count of sites; horizontal axis is QAF. Bin width = 0.02.\nGenotypes are color-coded: AA (yellow), AB (green), BB (blue).")

# VAF: All genotypes overlaid
VAF_all_gt <- ggplot(f2_homo_parents, aes(x = VAF, fill = GT)) +
  geom_histogram(binwidth = 0.02, alpha = 1, position = "identity") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.1)) +
  theme_classic() +
  labs(title = "A",
       x = "Variant Allele Frequency (VAF)",
       y = "Count") +
  scale_fill_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),
    name = "Genotype"
  ) +
  theme(legend.position = "none")

# VAF: Facetted by genotype
VAF_facet_gt <- ggplot(f2_homo_parents, aes(x = VAF, fill = GT)) +
  geom_histogram(binwidth = 0.02, alpha = 1, position = "identity") +
  facet_wrap(~ GT, scales = "free_y") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.1)) +
  theme_classic() +
  labs(title = "B",
       x = "Variant Allele Frequency (VAF)",
       y = "Count") +
  scale_fill_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),
    name = "Genotype"
  ) +
  theme(legend.position = "right")

# Combine VAF plots
VAF_combined <- (VAF_all_gt + VAF_facet_gt + plot_layout(widths = c(1, 3)))

# Display the combined VAF plot
VAF_combined

# Add caption below the plot
cat("**Figure 2.** Distribution of Variant Allele Frequency (VAF) values in F2 females from crosses with homozygous F0 and F1 parents.\nA. Overlaid histogram of VAF values across all genotypes.\nB. Faceted distribution of VAF by genotype, showing characteristic frequency patterns for AA, AB, and BB.\nVertical axis represents count of sites; horizontal axis is VAF. Bin width = 0.02.\nGenotypes are color-coded: AA (yellow), AB (green), BB (blue).")
```

#### short summary:
- 413,119 sites were homozygous in F0 and F1.    
- out of these sites, 45,841 sites were present in their corresponding families. Ofcourese, most of the sites are AA.    
- for these, we visualized the site's quality, in each genotype.    
- because we know what the QAF and VAF should be in a perfect genotyping, we can clearly see which of the sites are errors, and set the boundries accordingly.    

The plots make sense. Its visible that there's a peak of local quality heterozygotes below 0.4, which we can safely filter at.       
We therefore propose the following boundaries for each genotype:    
AA < 0.03     
AB > 0.4 and < 0.6   
BB > 0.8   

```{r create-filtered-df}
# Create the filtered sites dataset that will be used in the next Rmd file
filtered_sites <- all_param %>%
  filter(
    (GT == "AA" & QAF < 0.03 & VAF < 0.03) |   # Homozygous Reference (AA)
    (GT == "AB" & QAF > 0.4 & QAF < 0.6 & VAF > 0.4 & VAF < 0.6) |  # Heterozygous (AB)
    (GT == "BB" & QAF > 0.8 & VAF > 0.8)    # Homozygous Alternate (BB)
  ) %>%
  select(family, Sample, site, QAF, VAF, GT)  # Keep only necessary columns

# Save the filtered sites dataset for use in the next Rmd file
write.csv(filtered_sites, "/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/filtered_sites.csv", row.names = FALSE)

# Create summary statistics for filtered data
summary_stats <- filtered_sites %>%
  group_by(family, GT) %>%
  summarise(
    count = n(),
    mean_VAF = mean(VAF, na.rm = TRUE),
    mean_QAF = mean(QAF, na.rm = TRUE),
    .groups = 'drop'
  )

# Display summary statistics
knitr::kable(summary_stats, 
             caption = "Summary statistics for filtered sites by family and genotype",
             digits = 3)

# Save the f2_homo_parents data for use in the next Rmd file and create the QAF and VAF plots for the supp info of the MS
write.csv(f2_homo_parents, "/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/f2_homo_parents.csv", row.names = FALSE)
```



####  how many sites were kept after filtering?
```{r pull sites- spread to check, eval=FALSE, include=FALSE}

# Count the number of unique sites per sample before filtering
original_site_counts <- all_param %>%
 group_by(family, Sample) %>%  # Keep sites grouped by family
  summarise(original_sites = n_distinct(site), .groups = "drop") %>% 
  mutate(original_sites = as.numeric(original_sites)) %>%  
  mutate(family = as.character(family)) %>%
  mutate(Sample = as.character(Sample)) 

# Merge the original and filtered site counts by family
site_comparison <- original_site_counts %>%
  left_join(filtered_sites_summary, by = c("Sample", "family")) %>%
  mutate(kept_percentage = (num_sites_kept/original_sites) * 100)
  
ggplot(site_comparison, aes(x = family, y = kept_percentage, fill = family)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +  # Box plot without outlier points
  geom_jitter(width = 0.2, alpha = 0.5, size = 1) +  # Add jitter for visibility of individual points
  theme_minimal() +
  labs(
    title = "Distribution of Kept Site Percentages per Family",
    x = "Family",
    y = "Kept Site Percentage (%)",
    fill = "Family"
  ) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),  # Rotate x-axis for readability
    legend.position = "none"  # Remove legend to avoid redundancy
  )

```

---

# Follow genotype flow in one family, on one chromosome
Now we wish to follow the genotype in specific family, in specific chromosome.    
to do that, we need to look only at informative sites, for which we have hypothesis about the F2 genotype, based on their parents' (F1) genotype.   

Based on the determined site-quality threshold, we now filter the sites, and plot informative sites in one family on chromosome 1:    

```{r informative-sites-on-one-chrom}
# Step 1: Identify informative sites: where F0 (grandmother) = 0/0 and F1 (mother) = 0/1
F0AA_F1AB <- all_param %>%
  filter(generation %in% c("F0", "F1")) %>%  # Keep only F0 and F1 individuals
   filter(chrom %in% c("1")) %>%  # Keep only chromosome 1
 group_by(family, site) %>%  # Group by family and site
  summarise(
    grandmother_homo_ref = any(GT == "AA" & generation == "F0" & sex == "female"),
    mother_heterozygous = any(GT == "AB" & generation == "F1" & sex == "female"),
    .groups = "drop"
  ) %>%
  filter(grandmother_homo_ref & mother_heterozygous) %>%  # Ensure criteria are met
  select(family, site)  # Extract only relevant columns

# Step 2: Count the number of positions per family that passed the filtering
F0AA_F1AB %>%
  group_by(family) %>%
  summarise(num_sites_passed = n())

```


Now we have data on all families, will pick one family each time to plot
```{r one-family}
# set up family and members
fam_id <- 478

one_fam <- F0AA_F1AB %>% filter(family == fam_id)

# Step 3: Extract all family members' genotypes at these sites
family_sites <- one_fam %>%
  inner_join(filtered_sites, by = c("family", "site")) %>% # Keep only sites that passed filtering
  separate(site, into = c("chrom", "pos"), sep = "\\.1_", remove = FALSE) %>% # Split site into chrom and pos
  mutate(pos = as.numeric(pos)) %>% # Convert pos to numeric
  mutate(member = case_when(
    str_detect(Sample, "fnd") ~ "grandmother",
    str_detect(Sample, "_dat") ~ "mother",
    str_detect(Sample, "_son") ~ "father",
    str_detect(Sample, "_grndat") ~ "daughter",
    str_detect(Sample, "_grnson") ~ "son",
    TRUE ~ "unknown"
  )) %>%
  mutate(member = factor(member, levels = c("grandmother", "father", "mother", "son", "daughter"), ordered = TRUE)) %>%
  mutate(member_num = as.numeric(member))

# Find positions that appear in all members
positions_all_members <- family_sites %>%
  group_by(pos) %>%
  summarise(member_count = n_distinct(member), .groups = "drop") %>%
  filter(member_count == n_distinct(family_sites$member)) %>%
  pull(pos)

# Filter f2_filtered to keep only rows with these positions
fam2_common <- family_sites %>% filter(pos %in% positions_all_members)

# Filter f2_filtered to keep only sites with depth higher then 20
#fam2_common <- fam2_common %>% filter(DP > 30)

# Ensure positions are sorted
fam2_common <- fam2_common %>% arrange(pos)

# Generate 30 evenly spaced indices and round them to ensure integer indices
indices <- round(seq(1, nrow(fam2_common), length.out = 30))

# Select the corresponding positions
selected_positions <- fam2_common %>% slice(indices) %>% pull(pos)

# Filter the dataset based on these selected positions
fam2_selected <- fam2_common %>% filter(pos %in% selected_positions) %>%  filter(!is.na(member), !is.na(pos), !is.na(GT))
  
```


#### vizualize genotype flow in one family, for fig A1
```{r filter-site-qual-and-plot}
# Create a numeric x-axis for spacing (convert 'member' to factor and assign spacing)
fam2_selected %>%
 filter(
    (GT == "AA" & QAF < 0.03 & VAF < 0.03) |   # Homozygous Reference (AA)
    (GT == "AB" & QAF > 0.4 & QAF < 0.6 & VAF > 0.4 & VAF < 0.6) |  # Heterozygous (AB)
    (GT == "BB" & QAF > 0.8 & VAF > 0.8)    # Homozygous Alternate (BB)
  ) %>%
  ggplot(aes(x = member_num, y = pos / 1e6, color = GT)) +  # Convert pos to Mb
 geom_segment(aes(x = member_num - 0.3, xend = member_num + 0.3, 
                   y = pos / 1e6, yend = pos / 1e6), size = 1.5) +  # Apply same conversion
    scale_x_continuous(breaks = unique(fam2_selected$member_num), labels = unique(fam2_selected$member)) +  # Restore original labels
  scale_color_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),  # Updated colors for renamed genotypes
    name = "Genotype",
    breaks = c("AA", "AB", "BB")  # Ensures BB appears in the legend even if absent
  ) +
  theme_classic() +
  labs(
    title = paste("Genotype Flow Across Family Members 
Family ID:", fam_id),
    x = "Family Member",
    y = "Genomic 
Position (Mb)"
  ) +
  theme(
    axis.text.x = element_text(size = 12, face = "bold", angle = 30, hjust = 1),  # Rotate labels
    panel.grid.major = element_blank(), 
    axis.line.x = element_blank(),  # <-- Removes the x-axis line
    axis.ticks.x = element_blank())  # Removes x-axis ticks
```

save the filtered sites,to be used in the pulled data analysis       (~/Documents/GitHub/varroa-pedigree-study/R_scripts/After_revision/figs-for-varroa-pedigree-MS.Rmd)   
```{r}
# Filter sites based on QAF & VAF thresholds while keeping sites per family
filtered_sites <- all_param %>%
  group_by(family, site, Sample) %>%  # Keep sites grouped by family
  filter(
    (GT == "AA" & QAF < 0.03 & VAF < 0.03) |   # Homozygous Reference (AA)
    (GT == "AB" & QAF > 0.4 & QAF < 0.6 & VAF > 0.4 & VAF < 0.6) |  # Heterozygous (AB)
    (GT == "BB" & QAF > 0.8 & VAF > 0.8)    # Homozygous Alternate (BB)
  ) %>%
  ungroup() %>%  # Ensure the result is ungrouped for later use
  select(family, Sample, site, QAF, VAF, GT)  # Keep family-site-sample pairs without removing duplicates

filtered_sites_summary <- filtered_sites %>%
 group_by(family, Sample) %>%  # Keep sites grouped by family
  summarise(num_sites_kept = n(), .groups = "drop") %>%  # Keep counts by Sample
  mutate(family = as.character(family), 
         num_sites_kept = as.numeric(num_sites_kept))

# Save to CSV
write.csv(filtered_sites, "/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/filtered_sites.csv", row.names = FALSE)
```


```{r load-f2-filtered}
# save the fam2_selected to make the figure 1A in the figures RMD:  ~/Documents/GitHub/varroa-pedigree-study/R_scripts/After_revision/figs-for-varroa-pedigree-MS.Rmd
write.csv(fam2_selected, "/Users/nuriteliash/Documents/GitHub/varroa-pedigree-study/data/fam2_selected.csv", row.names = FALSE)
```


---
# Archived
#### A plot to vizualize gene flow in one family 
```{r eval=FALSE, include=FALSE}
# Create a numeric x-axis for spacing (convert 'member' to factor and assign spacing)
fam2_selected %>% 
  mutate(
    pos = ifelse(member %in% c("father", "mother"), pos + 100*10^6, pos),  # Adjusts positions
    pos = ifelse(member %in% c("grandmother"), pos + 100*10^6*2, pos)) %>%
  ggplot(aes(x = member_num, y = pos, color = GT)) +
  geom_segment(aes(x = member_num - 0.3, xend = member_num + 0.3, y = pos, yend = pos), size = 1.5) +  # Horizontal lines with spacing
  scale_x_continuous(breaks = unique(all_param$member_num), labels = unique(all_param$member)) +  # Restore original labels
  scale_color_manual(
    values = c("AA" = "#ffbf00", "AB" = "#66b032", "BB" = "#1982c4"),  # Updated colors for renamed genotypes
    name = "Genotype",
    breaks = c("AA", "AB", "BB")  # Ensures BB appears in the legend even if absent
  ) +
  theme_classic() +
  labs(
    title = paste("Genotype Flow Across Family Members 
Family ID:", fam_id),
    x = "Family Member",
    y = "Genomic Position"
  ) +
  theme(
    axis.text.x = element_text(size = 12, face = "bold", angle = 30, hjust = 1),  # Rotate labels
    panel.grid.major = element_blank(), 
    axis.line.x = element_blank(),  # <-- Removes the x-axis line
    axis.ticks.x = element_blank())  # Removes x-axis ticks
```

